pinot-common/src/main/java/org/apache/pinot/common/metadata/ZKMetadataProvider.java

```diff
@@ -73,6 +79,87 @@ public static void setUserConfig(ZkHelixPropertyStore<ZNRecord> propertyStore, S
     propertyStore.set(constructPropertyStorePathForUserConfig(username), znRecord, AccessOption.PERSISTENT);
   }
 
+  /**
+   * Create database config, fail if exists.
+   *
+   * @return true if creation is successful.
+   */
+  public static boolean createDatabaseConfig(
+      ZkHelixPropertyStore<ZNRecord> propertyStore, DatabaseConfig databaseConfig) {
+    String databaseName = databaseConfig.getDatabaseName();
+    String databaseConfigPath = constructPropertyStorePathForDatabaseConfig(databaseName);
+    ZNRecord databaseConfigZNRecord;
+    try {
+      databaseConfigZNRecord = toZNRecord(databaseConfig);
+    } catch (Exception e) {
+      LOGGER.error("Caught exception constructing ZNRecord from database config for database: {}", databaseName, e);
+      return false;
+    }
+    return propertyStore.create(databaseConfigPath, databaseConfigZNRecord, AccessOption.PERSISTENT);
+  }
+
+  /**
+   * Update database config.
+   *
+   * @return true if update is successful.
+   */
+  public static boolean setDatabaseConfig(ZkHelixPropertyStore<ZNRecord> propertyStore, DatabaseConfig databaseConfig) {
+    String databaseName = databaseConfig.getDatabaseName();
+    ZNRecord databaseConfigZNRecord;
+    try {
+      databaseConfigZNRecord = toZNRecord(databaseConfig);
+    } catch (Exception e) {
+      LOGGER.error("Caught exception constructing ZNRecord from database config for database : {}", databaseName, e);
+      return false;
+    }
+    return propertyStore.set(constructPropertyStorePathForDatabaseConfig(databaseName), databaseConfigZNRecord,
+        -1, AccessOption.PERSISTENT);
+  }
+
+  /**
+   * Remove database config.
+   */
+  @VisibleForTesting
+  public static void removeDatabaseConfig(ZkHelixPropertyStore<ZNRecord> propertyStore, String databaseName) {
+    propertyStore.remove(constructPropertyStorePathForDatabaseConfig(databaseName), AccessOption.PERSISTENT);
+  }
+
+  private static ZNRecord toZNRecord(DatabaseConfig databaseConfig) {
```

Comment:
does this method toZNRecord throw Exception? as it's just put() calls inside. If not, then can remove the try-catch blocks in the methods above

Reply:
You are right, will remove it.